distribution/lib/Standard/Base/0.0.0-dev/src/Network/Reload_Detector.enso

```diff
@@ -0,0 +1,32 @@
+import Standard.Base.Data.Boolean.Boolean
+import Standard.Base.Nothing.Nothing
+import Standard.Base.Runtime.Managed_Resource.Managed_Resource
+import Standard.Base.Runtime.Ref.Ref
+
+## PRIVATE
+   This is used by ReloadDetector.java to create a `Managed_Resource` that is
+   finalized when the reload button is pressed.
+
+   The `on_finalize` function and the `clear` method both write `Nothing` to the
+   ref. This is a signal that a reload has happenend. `on_finalize` is called by
+   the engine when a reload happens. `clear` is only for testing, to simulate a
+   reload.
+
+   The `0` value stored in the ref is not used; it just has to be something
+   other than Nothing.
+type Reload_Detector
+    private Value mr:Managed_Resource
+
+    new -> Reload_Detector =
+        ref = Ref.new 0
+        on_finalize ref = ref.put Nothing
+        mr = Managed_Resource.register ref on_finalize Boolean.True
+        Reload_Detector.Value mr
+
+    get self = self.mr.with .get
+
+    clear self =
+        self.mr.with (ref-> ref.put Nothing)
```

Comment:
Given the [most recent changes in the implementation](https://github.com/enso-org/enso/pull/11577#discussion_r1867063485) this code shouldn't set the `ref` to `Nothing`, but to a `DataflowError` - however I admit, I have no way to do it! I don't seem to be able to send `Error` into `ref.put` and I even managed to crash the engine while doing that #11774

Reply:
I am using a special value for the test-only implementation.